starmark.dev/src/utils/cssesc.mjs

```
@@ -0,0 +1,89 @@
+  while (counter < length) {
+    const character = string.charAt(counter++);
+    const codePoint = character.charCodeAt();
+    let value;
+    
+    // Handle non-printable ASCII characters
+    if (codePoint < 0x20 || codePoint > 0x7E) {
+      if (codePoint >= 0xD800 && codePoint <= 0xDBFF && counter < length) {
+        const extra = string.charCodeAt(counter++);
+        if ((extra & 0xFC00) == 0xDC00) {
+          const combinedCodePoint = ((codePoint & 0x3FF) << 10) + (extra & 0x3FF) + 0x10000;
+          value = '\\' + combinedCodePoint.toString(16).toUpperCase() + ' ';
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)